added star formation and SN feedback test

pocvirk opened the pull request:
This PR uses the cosmo test, runs it with AMR until stars form (model 0), and then a couple of SN explode (kinetic model).
A new namelist parameter is added to prevent random Poisson noise to be added to the star. This was needed to be able to reproduce the results on different machines.

pocvirk committed:
added star formatino and SN feedback test

tinecolman commented:
We will need to add the new star category to the test workflow for it to run on github actions.

pocvirk committed:
added a parameter randomize_sf=.true. by default but can be set to .false. for tests where randomization is to be avoided. Impacts star_formation.f90, read_hydro_parameters and amr_parameters. Also, added a star/sfonly dir where we just test star formation, not SN FB. Then SNFB is tested in its own dir, also with randomize_sf=.false.

pocvirk commented:
to deal with the rather large errors in this test:
I added a parameter randomize_sf=.true. by default but can be set to .false. for tests where randomization is to be avoided. Impacts star_formation.f90, read_hydro_parameters and amr_parameters. Also, added a star/sfonly dir where we just test star formation, not SN FB. Then SNFB is tested in its own dir, also with randomize_sf=.false."

pocvirk committed:
updated the nstar(i)=1 to nstar(i)=PoissMean, updated the reference solutions accordingly

pocvirk committed:
removed sfonly from PR

pocvirk committed:
edited SNFB.nml to suggest no-FB usage and tests.yaml to activate the star tests

pocvirk commented:
removed tests/star/sfonly from tracking and repo.

tinecolman commented:
<img width="791" height="595" alt="Screenshot from 2025-10-02 14-53-10" src="https://github.com/user-attachments/assets/51305591-97b9-4a1e-a67c-6635f2388e99" />

This is what it looks like.

tinecolman committed:
Merge branch 'dev' into star_SN_FB

tinecolman committed:
alphabetical order

tinecolman committed:
make FB test work with restart test mechanism
